shambomon/assets/js/shambomon.jsx

```diff
@@ -96,37 +96,34 @@ class Shambomon extends React.Component {
     return ready;
   }
 
-  // Returns the id of the winner; else returns false
-  hasWinner() {
+  // Returns the id of the winner
+  getWinner() {
     let players = this.state.players;
     // Player 2 wins
     if (players[0].health <= 0) {
       return players[1].id;
     }
     // Player 1 wins
-    else if (players[1].health <= 0) {
-      return players[0].id;
-    }
-    // No one has won yet
     else {
-      return false;
+      return players[0].id;
     }
   }
 
   // Main render function
   render() {
+    console.log(this.state)
     let ready = this.isReady();
-    let winner = this.hasWinner();
     // Game has less than two players
-    if (!ready) {
+    if (!ready && !this.state.gameOver) {
       return <Waiting />;
     }
     // Someone has won
-    else if (winner) {
+    else if (this.state.gameOver) {
```

Comment:
`this.state.gameOver` gets set to `true` on the server if either player's HP is <= 0. Before, this check was done on the browser.